### PR TITLE
Change return type for conditional correlation and covariance

### PR DESCRIPTION
**Pull request by adaluong**
* Returns a DataFrame instead of a ndarray if input Y is a DataFrame
* Includes labels from Y

### CODE REVIEW

File: cinspect/stats.py [hunk 1]
```diff
@@ -55,6 +56,10 @@ def conditional_cov(X, Y, estimator=None, bias=False, ddof=None):
     EY_X = estimator.fit(X, Y).predict(X)
     RY = Y - EY_X
     cov = np.cov(RY.T, bias=bias, ddof=ddof)  # equal E[(Y-E[Y|X])(Y-E[Y|X]).T]
+
+    if isinstance(Y, pd.DataFrame):
+        cov = _ndarray_to_df(cov, Y)
```
Comment (dsteinberg) [inline cinspect/stats.py:61]: Very neat - I like this :-)

File: tests/test_stats.py [hunk 2]
```diff
@@ -54,6 +55,8 @@ def test_conditional_correlation_correctness(func, ccorr, corr):
     """Test the correctness of the conditional correlation computation."""
     X, Y = func()
     calc_ccorr = conditional_corrcoef(X, Y)
+    if (isinstance(calc_ccorr, pd.DataFrame)):
+        calc_ccorr = calc_ccorr.to_numpy()
```
Comment (dsteinberg) [inline tests/test_stats.py:59]: We may not need this since we control the type going into this test

File: tests/test_stats.py [hunk 3]
```diff
@@ -68,6 +71,8 @@ def test_inputs(ysize, xsize):
     X = np.random.randn(N, xsize)
 
     ccov = conditional_cov(X, Y)
+    if (isinstance(ccov, pd.DataFrame)):
+        ccov = ccov.to_numpy()
```
Comment (dsteinberg) [inline tests/test_stats.py:75]: See my previous comment.

Also, what may be worth doing, either in this function or later, is to explicitly test that if Y is a dataframe, that return type is a dataframe.

You could do something like:
```
Ydf = pd.DataFrame(Y, columns=["y", "z"])
... run the functions and test for the right output and column/index names
```

(I wouldn't bother re-testing the correct values are returned)

**Review by dsteinberg — CHANGES_REQUESTED** [review]
Looks really good!
The only thing I would suggest is to explicitly test your type conversion code, see my inline comments for some suggestions.

**Comment by adaluong** [comment]
I've modified the function to only convert to a df if there's more than one column in Y and have added tests as requested @dsteinberg

**Review by dsteinberg — APPROVED** [review]
LGTM!

**Pull request merged**